// @shopify/polaris-tokens/dist/cjs/src/themes/base/index.js
'use strict';

var utilities = require('../../utilities.js');
var border = require('./border.js');
var breakpoints = require('./breakpoints.js');
var color = require('./color.js');
var font = require('./font.js');
var height = require('./height.js');
var motion = require('./motion.js');
var shadow = require('./shadow.js');
var space = require('./space.js');
var text = require('./text.js');
var width = require('./width.js');
var zIndex = require('./zIndex.js');

var createMetaThemeBase = utilities.createExact();
var metaThemeBase = createMetaThemeBase({
  border: utilities.tokensToRems(border.border),
  breakpoints: utilities.tokensToRems(breakpoints.breakpoints),
  color: color.color,
  font: utilities.tokensToRems(font.font),
  height: utilities.tokensToRems(height.height),
  motion: motion.motion,
  shadow: utilities.tokensToRems(shadow.shadow),
  space: utilities.tokensToRems(space.space),
  text: text.text,
  width: utilities.tokensToRems(width.width),
  zIndex: zIndex.zIndex
});

exports.metaThemeBase = metaThemeBase;
